### PR TITLE
TT: recycler view adapter unit tests

### PR DESCRIPTION
**Pull request by githengi**

### CODE REVIEW

File: opensrp-app/src/test/java/org/smartregister/cursoradapter/RecyclerViewCursorAdapterTest.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        when(cursor.moveToPosition(1)).thenReturn(false);
+        assertEquals(0, recyclerViewCursorAdapter.getItemId(1));
+        verify(cursor).moveToPosition(1);
+        verify(cursor, never()).getLong(anyInt());
+    }
+
+    @Test
+    public void testSetHasStableIdsShouldPassTrueToSuper() {
+        recyclerViewCursorAdapter.setHasStableIds(false);
+        assertEquals(true, ReflectionHelpers.getField(recyclerViewCursorAdapter, "mHasStableIds"));
```
Comment (vincent-karuri) [inline opensrp-app/src/test/java/org/smartregister/cursoradapter/RecyclerViewCursorAdapterTest.java:86]: Can this be simplified to use `assertTrue`? Also, how comes this remains true yet you set it to false above?

File: opensrp-app/src/test/java/org/smartregister/cursoradapter/RecyclerViewCursorAdapterTest.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+        recyclerViewCursorAdapter.setHasStableIds(true);
+        assertEquals(true, ReflectionHelpers.getField(recyclerViewCursorAdapter, "mHasStableIds"));
```
Comment (vincent-karuri) [inline opensrp-app/src/test/java/org/smartregister/cursoradapter/RecyclerViewCursorAdapterTest.java:89]: Can this be simplified to use `assertTrue`?

Comment (githengi) [reply opensrp-app/src/test/java/org/smartregister/cursoradapter/RecyclerViewCursorAdapterTest.java]: done

File: opensrp-app/src/test/java/org/smartregister/cursoradapter/RecyclerViewCursorAdapterTest.java (hunk 1 above)
Comment (githengi) [reply opensrp-app/src/test/java/org/smartregister/cursoradapter/RecyclerViewCursorAdapterTest.java]: done

**Review by vincent-karuri — APPROVED** [review]

**Pull request merged**